### PR TITLE
LPS-133418 | master

### PR DESCRIPTION
**Pull request by ThalitaCeli**

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase [hunk 1]
```diff
@@ -98,4 +98,59 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-124704. Asset Publisher view.jsp has potential XSS vulnerability."
+	@priority = "4"
+	@uitest
+	test ViewAssetEntryIdInAssetPublisher {
+		property testray.component.names = "Web Content Administration";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase:105]: Unneeded

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase [hunk 2]
```diff
@@ -98,4 +98,59 @@ definition {
...
+				layoutName = "Asset Publisher Page",
+				widgetName = "Asset Publisher");
+
+			Navigator.gotoSitePage(
+				pageName = "Asset Publisher Page",
+				siteName = "Test Site Name");
+
+			AssertVisible(
+				key_assetTitle = "WC WebContent Title ${i}",
+				locator1 = "AP#ASSET_ABSTRACT_TITLE");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase:132]: Redundant

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase [hunk 3]
```diff
@@ -98,4 +98,59 @@ definition {
...
+		}
+
+		task ("Click on one of the web content titles in Asset Publisher") {
+			AssetPublisherPortlet.clickAssetTitle(assetTitle = "WC WebContent Title ${i}");
+		}
+
+		task ("Navigate back to the content page") {
+			PageEditor.clickBack();
+		}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase:141]: Content page doesn't exist. Should click the back button on Asset Publisher

**Review by Tim-Cao — COMMENTED** [review]
Hi @ThalitaCeli, Could you move your case to AssetPubilsherWithXSS.testcase? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase [hunk 4]
```diff
@@ -98,4 +98,59 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-124704. Asset Publisher view.jsp has potential XSS vulnerability."
+	@priority = "4"
+	@uitest
+	test ViewAssetEntryIdInAssetPublisher {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase:104]: ViewAssetDetailsWhenAppendXSSToURL

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase (hunk 2 above)
Comment (ThalitaCeli) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/webcontentdisplay/WebContentDisplayWithXSS.testcase:132]: OK. Does this need some AssertVisible?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+			PortalInstances.tearDownCP();
+		}
+		else {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "test@liferay.com",
+				userLoginFullName = "Test Test");
+
+			Site.tearDownCP();
+
+			JSONUser.tearDownNonAdminUsers();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase:38]: Use `JSONGroup.deleteGroupByName(groupName = "Test Site Name");`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+		task ("Add a page with an Asset Publisher widget and display the Web Contents article in the Asset Publisher Widget") {
+			JSONLayout.addPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Asset Publisher Page");
+
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Asset Publisher Page",
+				widgetName = "Asset Publisher");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase:63]: Duplicate steps with set up block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+		}
+
+		task ("Click the back button on Asset Publisher") {
+			PageEditor.clickBack();
+		}
+
+		task ("View in URL AssetEntryId Validate") {
+			var portalURL = PropsUtil.get("portal.url");
+
+			var assetEntryId = '''
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase:81]: Had better to get URL from browser then replace assetEntryId value

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+				userLoginFullName = "Test Test");
+
+			Site.tearDownCP();
+
+			JSONUser.tearDownNonAdminUsers();
+		}
+	}
+
+
+	@description = "This is a test for LPS-124704. Asset Publisher view.jsp has potential XSS vulnerability."
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase:42]: No alert popup on view asset in Asset Publisher when replace the value of assetEntryId by XSS in URL.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+				pageName = "Asset Publisher Page",
+				siteName = "Test Site Name");
+		}
+
+		task ("Click on one of the web content titles in Asset Publisher") {
+			AssetPublisherPortlet.clickAssetTitle(assetTitle = "WC WebContent Title ${i}");
+		}
+
+		task ("Click the back button on Asset Publisher") {
+			PageEditor.clickBack();
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/asset/assetpublisher/AssetPublisherWithXSS.testcase:75]: Navigator.gotoBack();

**Comment by Tim-Cao** [comment]
Hi @ThalitaCeli, see changes https://github.com/liferay-echo/liferay-portal/compare/885d96207ee621c685fda04d8630f6f8e9e20f60...d4a7b48cbf7c95c04417a614e3077dec006f65e7

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**